src/node.rs

```diff
@@ -330,30 +336,33 @@ impl Node {
         self.machine.current().chain()
     }
 
-    /// Returns the underlying Elder state.
-    pub fn elder_state(&self) -> Option<&crate::states::Elder> {
-        self.machine.current().elder_state()
+    /// Returns the underlying ApprovedPeer state.
+    pub fn approved_peer_state(&self) -> Option<&crate::states::ApprovedPeer> {
+        self.machine.current().approved_peer_state()
     }
 
-    /// Returns mutable reference to the underlying Elder state.
-    pub fn elder_state_mut(&mut self) -> Option<&mut crate::states::Elder> {
-        self.machine.current_mut().elder_state_mut()
+    /// Returns mutable reference to the underlying ApprovedPeer state.
+    pub fn approved_peer_state_mut(&mut self) -> Option<&mut crate::states::ApprovedPeer> {
+        self.machine.current_mut().approved_peer_state_mut()
     }
 
-    /// Returns the underlying Elder state unwrapped - panics if not Elder.
-    pub fn elder_state_unchecked(&self) -> &crate::states::Elder {
-        unwrap!(self.elder_state(), "Should be State::Elder")
+    /// Returns the underlying ApprovedPeer state unwrapped - panics if not ApprovedPeer.
+    pub fn approved_peer_state_unchecked(&self) -> &crate::states::ApprovedPeer {
+        self.approved_peer_state()
+            .expect("should be State::ApprovedPeer")
     }
 
-    /// Returns whether the current state is `Elder`.
+    /// Returns whether the node is Elder.
     pub fn is_elder(&self) -> bool {
-        self.elder_state().is_some()
+        self.chain()
+            .map(|chain| chain.is_self_elder())
+            .unwrap_or(false)
     }
 
-    /// Returns whether the current state is `Elder` or `Adult`.
+    /// Returns whether the node is Elder or Adult.
```

Comment:
We should consider Infant as well I reckon, although infants should not cause us to work (very much), so no parsec and not in our members.